samples/browser/pub_sub/index.ts

```diff
@@ -13,39 +13,163 @@ function log(msg: string) {
     $('#console').append(`<pre>${msg}</pre>`);
 }
 
-async function fetch_credentials() {
-    return new Promise<AWS.CognitoIdentityCredentials>((resolve, reject) => {
-        AWS.config.region = Config.AWS_REGION;
-        const credentials = AWS.config.credentials = new AWS.CognitoIdentityCredentials({
-            IdentityPoolId: Config.AWS_COGNITO_IDENTITY_POOL_ID
+
+/**
+ * AWSCognitoCredentialOptions. The credentials options used to create AWSCongnitoCredentialProvider.
+ *
+ * @category Auth
+ */
+class AWSCognitoCredentialOptions implements auth.CredentialsProviderOptions
+{
+    IdentityPoolId : string;
+    Region: string;
+
+    constructor(IdentityPoolId: string, region: string)
```

Comment:
I think you can just put `'private` in front of the parameter names and it will automatically turn them into properties on the class and they'll be initialized properly without explicitly assigning.